osu.Game/Screens/Edit/Components/SummaryTimeline.cs

```diff
@@ -0,0 +1,357 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Linq;
+using OpenTK;
+using osu.Framework.Allocation;
+using osu.Framework.Configuration;
+using osu.Framework.Extensions.IEnumerableExtensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Beatmaps.Timing;
+using osu.Game.Graphics;
+
+namespace osu.Game.Screens.Edit.Components
+{
+    /// <summary>
+    /// The timeline that sits at the bottom of the editor.
+    /// </summary>
+    public class SummaryTimeline : CompositeDrawable
+    {
+        private const float corner_radius = 5;
+        private const float contents_padding = 15;
+        private const float marker_bar_width = 2;
+
+        private readonly Drawable background;
+
+        private readonly Container markerContainer;
+
+        private readonly Drawable timelineBar;
+        private readonly Drawable marker;
+
+        private readonly Bindable<WorkingBeatmap> beatmap = new Bindable<WorkingBeatmap>();
+
+        public SummaryTimeline()
+        {
+            Masking = true;
+            CornerRadius = 5;
+
+            InternalChildren = new[]
+            {
+                    background = new Box { RelativeSizeAxes = Axes.Both },
+                    new Container
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Padding = new MarginPadding { Left = contents_padding, Right = contents_padding },
+                        Children = new[]
+                        {
+                            markerContainer = new Container

```

Comment:
i'd probably make this its own class

Reply:
I also thing this should be on top of everything else visually

Reply:
Re: Making marker on top of everything: I'm going to keep it as it is for now since this was flyte's previous design and this is going to change anyway.